gradle-baseline-java/src/main/java/com/palantir/baseline/plugins/BaselineModuleJvmArgs.java

```diff
@@ -406,8 +326,84 @@ static Builder builder() {
         class Builder extends ImmutableJarManifestModuleInfo.Builder {}
     }
 
-    enum OpensMode {
-        COMPILATION,
-        RUNTIME;
+    private static final class ModuleJvmArgsArgumentProvider implements CommandLineArgumentProvider, Serializable {
```

Comment:
I think here, you could make this an `abstract` class and then replace `Provider<Set<String>> exports` with `SetProperty<String> getExports()`. Since the java compile tasks lists these as `@Nested`:

```java
@Nested
public List<CommandLineArgumentProvider> getCompilerArgumentProviders() {
    return this.compilerArgumentProviders;
}
```

it means that we can also annotate thense all as `@Input`s and if there were task dependencies (say from the exports/opens - unlikely though) they would get linked up automatically.

Also makes it Serializable automatically.

Also reduces some boilerplate and allows injecting things.